client/lecturer/app/lecturer.controller.js

```diff
@@ -6,17 +6,20 @@
         .controller('LecturerCtrl', LecturerController);
     
     /* @ngInject */
-    function LecturerController($scope, $location, userFactory) {
+    function LecturerController($scope, lecturerFactory) {
         console.log('Ready (Lecturer Controller)');
         
         var vm = this;
-        
-        $scope.$on("$routeChangeStart", function(event, next, current) {
-            console.log('Route change (Lecture Controller)');
+
+        $scope.$on('locationChangeStart', function() {
+            console.log('location change');
             if (angular.isDefined(next)  &&  angular.isDefined(next.$$route)) {
-                if (!userFactory.isLoggedIn()  &&  next.$$route.loginRequired) {
-                    event.preventDefault();
-                    $location.path('/');
+                
+                if (next.$$.route.loginRequired) {
+                    lecturerFactory.checkUserToken(function() {
+                        // Prevent access to next route on error.
+                        event.preventDefault();
```

Comment:
The event is passed from routeChangeStart as parameter into the anonymous function.